tests/test_graph_components/providers/test_nlu_training_data_provider.py

```diff
@@ -51,4 +51,24 @@ def test_nlu_training_data_provider(
             str(resource_directory), DEFAULT_TRAINING_DATA_OUTPUT_PATH
         )
         data = load_data(resource_name=data_file, language="en")
+        assert os.path.isfile(data_file)
         assert isinstance(data, TrainingData)
+
+        # delete the persisted data
+        os.remove(data_file)
+        assert not os.path.isfile(data_file)
+
+    # create a provider with persist == False
+    NLUTrainingDataProvider.create(
+        {"language": "en", "persist": False},
+        default_model_storage,
+        resource,
+        default_execution_context,
+    )
+
+    # check the data was not persisted
```

Comment:
Don't we need to call `provide` to check if this really isn't persisting?
Maybe we could paramaterize the test?

Reply:
bump

Reply:
Oh yeah you're right.

Reply:
Okay, now fixed!